completion/available/maven.completion.bash

```diff
@@ -24,7 +25,7 @@ _mvn() {
 	# Work-around borrowed from the darcs work-around for the same
 	# issue.
 	colonprefixes=${cur%"${cur##*:}"}
-	COMPREPLY=($(compgen -W '$cmds' -- $cur))
+	COMPREPLY=($(compgen -W "$cmds" -- "$cur"))
 	local i=${#COMPREPLY[*]}
 	while [ $((--i)) -ge 0 ]; do
```

Comment:
```suggestion
	while [ "$((--i))" -ge 0 ]; do
```

I'm not sure why shellcheck doesn't complain about this, but even if the result is ensured to be a number, it needs to be quoted for a custom `IFS`.

Reply:
no clue why this happens. there are several places where it doesn't complain where I expect it to. sometimes I fix them without it asking, like declaring local and assigning on the same line. it mainly hates it if the value is a subshell $()

Reply:
Maybe shellcheck doesn't care about custom IFS as far as IFS is not modified in the shell script. However, the shell interactive settings should care about the user's custom shell settings. Maybe the special situation with shell interactive settings is something that shellcheck doesn't cover.

Reply:
when it starts understanding the code on that level, lots of people WILL lose jobs :)